core/src/main/src/ProcessRunner/StreamReader.java

```diff
@@ -0,0 +1,63 @@
+package ProcessRunner;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.function.Consumer;
+
+/**
+ * Created by Virtuality.
+ */
+
+//Class is largely based on:
+//https://stackoverflow.com/questions/14165517/processbuilder-forwarding-stdout-and-stderr-of-started-processes-without-blocki
+class StreamReader extends Thread {
+
+    private InputStream inputStream;
+    private Consumer<String> onInputRead;
+
+    StreamReader(InputStream inputStream) {
+        this.inputStream = inputStream;
+    }
+
+
+    @Override
+    public void run() {
+        try {
+            readCharLoop();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    void setOnInputRead(Consumer<String> consumer) {
+        this.onInputRead = consumer;
+    }
+
+
+    private void readCharLoop() throws IOException {
+        InputStreamReader isr = new InputStreamReader(inputStream);
+        BufferedReader br = new BufferedReader(isr);
+
+        int receivedValue = 0;
+        while (receivedValue != -1) {
+            receivedValue = br.read();
+            if (receivedValue != 0) {
+                onInputRead.accept(String.valueOf(((char)receivedValue)));
+            }
+        }
+        System.out.println("StreamReader closed!");
+    }
+
+
+    //Bug found, will only return a string when a \n has been read.
```

Comment:
Is it necessary when we have a char reader? If so, feel free to comment the purpose and UC of the method 😄